packages/logging_cloudwatch/aws_logging_cloudwatch/lib/src/remote_constraint_provider.dart

```diff
@@ -33,9 +37,83 @@ class DefaultRemoteLoggingConstraintProvider
   final DefaultRemoteConfiguration _config;
   final AWSCredentialsProvider _credentialsProvider;
 
+  LoggingConstraint? _loggingConstraint;
+  DateTime? _lastUpdated;
+
+  Future<void> _saveConstraintLocally(LoggingConstraint constraint) async {
+    final file = File('logging_constraint.json');
+    await file.writeAsString(jsonEncode(constraint));
+  }
+
+  Future<LoggingConstraint?> _getConstraintFromLocalStorage() async {
+    final file = File('logging_constraint.json');
+    if (await file.exists()) {
+      final content = await file.readAsString();
+      return LoggingConstraint.fromJson(
+        jsonDecode(content) as Map<String, dynamic>,
+      );
+    }
+    return null;
+  }
+
+  Future<void> _fetchAndCacheConstraintFromEndpoint() async {
+    try {
+      final constraint = await _fetchConstraintFromEndpoint();
```

Comment:
this wait on saving to local file and then load it to cache. better to load to cache and then save to local file. maybe _fetchConstraintFromEndpoint() should only fetch from endpoint and not save it to local file. saving to local file can be done in this function after loading it to cache.

Reply:
So you're saying in the fetchConstraintFromEndpoint method, I should remove the _saveConstraintLocally() method from inside of there and separate it so that it's called separately?